src/lib/state-transitions.js

```diff
@@ -11,12 +11,26 @@ module.exports = (definition) => {
   // targets for states outside the containers.
   const nextAndDefaultTargets = (nestedStateMachine) => {
     const states = [];
-    Object.keys(nestedStateMachine.States).forEach((stateName) => {
-      const nestedState = nestedStateMachine.States[stateName];
-      const isContainer = ['Map', 'Parallel'].indexOf(nestedState.Type) >= 0;
-      const path = isContainer ? '$.[Next,Default]' : '$..[Next,Default]';
-      states.push(...JSONPath({ json: nestedState, path }));
-    });
+    Object.keys(nestedStateMachine.States)
+      .forEach((stateName) => {
+        const nestedState = nestedStateMachine.States[stateName];
+        const isContainer = ['Map', 'Parallel'].indexOf(nestedState.Type) >= 0;
+        if (isContainer) {
+          states.push(...JSONPath({
+            json: nestedState,
+            path: '$.Next',
+          }));
+          states.push(...JSONPath({
+            json: nestedState,
+            path: '$.Default',
+          }));
+        } else {
+          states.push(...JSONPath({
+            json: nestedState,
+            path: '$..[Next,Default]',
+          }));
+        }
```

Comment:
good test case and change

It looks like the JSONPath expression `$.[Next,Default]` is descending into the nested branch when it wasn't expected to. Unclear if this is the expected behavior in the jsonpath plus library but it's not what we want here and leads to the false positive.

Then other changes look like whitespace only.